File: go/marsroverkata/sample5_test.go
package marsrover

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestFollowAllInstructions(t *testing.T) {
	plateau := Plateau{maxX: 5, maxY: 5}
	startingPosition := Coordinates{1,2}
	marsRover := MarsRover{plateau: plateau, heading: N, position: startingPosition}

	commands := []Command{B, F, L, F, F, R}
	for _, command := range commands {
		switch command {
		case B:
			marsRover.backward()
		case F:
			marsRover.forward()
		case L:
			marsRover.turnLeft()
		case R:
			marsRover.turnRight()
		}
	}
	expectedPosition := Coordinates{-1,2}
	assert.Equal(t, expectedPosition, marsRover.coordinates())
	assert.Equal(t, N, marsRover.heading)
}

File: javascript/sample5.test.js
import Plateau from "./plateau";
import MarsRover from "./mars-rover";

describe('Mars Rover', function() {

    it('can follow instructions', function () {
        var plateau = new Plateau(5, 5);
        var rover = new MarsRover(plateau, [1, 2], "N");

        var commands = ["B", "F", "L", "F", "F", "R"]
        commands.forEach(function (command, index) {
            switch (command) {
                case "B":
                    rover.backward();
                case "F":
                    rover.forward();
                case "L":
                    rover.turnLeft();
                case "R":
                    rover.turnRight();
            }
        })

        expect(rover.position).toEqual([-1, 2]);
        expect(rover.heading).toEqual("N");

    });

});